File: conformance/code/individual/src/main/java/com/digit/individual/model/AddressDTO.java
package com.digit.individual.model;

import com.fasterxml.jackson.annotation.JsonInclude;

/** Wire shape for an address. Mirrors Go internal/models/address_dto.go. */
@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class AddressDTO {
    private String id = "";
    private String type;
    private String doorNo;
    private String buildingName;
    private String street;
    private String landmark;
    private String addressLine1;
    private String addressLine2;
    private String city;
    private String region;
    private String country;
    private String pincode;
    private String boundaryCode;
    private Double latitude;
    private Double longitude;
    private Double locationAccuracy;
    private String requestId;
    private AuditDetail auditDetail;

    @JsonInclude(JsonInclude.Include.ALWAYS)
    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    public String getType() { return type; }
    public void setType(String type) { this.type = type; }
    public String getDoorNo() { return doorNo; }
    public void setDoorNo(String doorNo) { this.doorNo = doorNo; }
    public String getBuildingName() { return buildingName; }
    public void setBuildingName(String buildingName) { this.buildingName = buildingName; }
    public String getStreet() { return street; }
    public void setStreet(String street) { this.street = street; }
    public String getLandmark() { return landmark; }
    public void setLandmark(String landmark) { this.landmark = landmark; }
    public String getAddressLine1() { return addressLine1; }
    public void setAddressLine1(String addressLine1) { this.addressLine1 = addressLine1; }
    public String getAddressLine2() { return addressLine2; }
    public void setAddressLine2(String addressLine2) { this.addressLine2 = addressLine2; }
    public String getCity() { return city; }
    public void setCity(String city) { this.city = city; }
    public String getRegion() { return region; }
    public void setRegion(String region) { this.region = region; }
    public String getCountry() { return country; }
    public void setCountry(String country) { this.country = country; }
    public String getPincode() { return pincode; }
    public void setPincode(String pincode) { this.pincode = pincode; }
    public String getBoundaryCode() { return boundaryCode; }
    public void setBoundaryCode(String boundaryCode) { this.boundaryCode = boundaryCode; }
    public Double getLatitude() { return latitude; }
    public void setLatitude(Double latitude) { this.latitude = latitude; }
    public Double getLongitude() { return longitude; }
    public void setLongitude(Double longitude) { this.longitude = longitude; }
    public Double getLocationAccuracy() { return locationAccuracy; }
    public void setLocationAccuracy(Double locationAccuracy) { this.locationAccuracy = locationAccuracy; }
    public String getRequestId() { return requestId; }
    public void setRequestId(String requestId) { this.requestId = requestId; }
    public AuditDetail getAuditDetail() { return auditDetail; }
    public void setAuditDetail(AuditDetail auditDetail) { this.auditDetail = auditDetail; }
}

File: conformance/code/individual-go/internal/models/address_dto.go
package models

// AddressDTO is the wire shape per v3 spec.
// Notably absent vs. the entity: tenantId, clientReferenceId, formatted,
// wardCode (the latter three were dropped entirely in wave C).
type AddressDTO struct {
	ID               string   `json:"id"`
	Type             string   `json:"type,omitempty"`
	DoorNo           string   `json:"doorNo,omitempty"`
	BuildingName     string   `json:"buildingName,omitempty"`
	Street           string   `json:"street,omitempty"`
	Landmark         string   `json:"landmark,omitempty"`
	AddressLine1     string   `json:"addressLine1,omitempty"`
	AddressLine2     string   `json:"addressLine2,omitempty"`
	City             string   `json:"city,omitempty"`
	Region           string   `json:"region,omitempty"`
	Country          string   `json:"country,omitempty"`
	Pincode          string   `json:"pincode,omitempty"`
	BoundaryCode     string   `json:"boundaryCode,omitempty"`
	Latitude         *float64 `json:"latitude,omitempty"`
	Longitude        *float64 `json:"longitude,omitempty"`
	LocationAccuracy *float64 `json:"locationAccuracy,omitempty"`

	RequestID   string       `json:"requestId,omitempty"`
	AuditDetail *AuditDetail `json:"auditDetail,omitempty"`
}

func AddressFromEntity(e *Address) *AddressDTO {
	if e == nil {
		return nil
	}
	return &AddressDTO{
		ID:               e.ID,
		Type:             e.Type,
		DoorNo:           e.DoorNo,
		BuildingName:     e.BuildingName,
		Street:           e.Street,
		Landmark:         e.Landmark,
		AddressLine1:     e.AddressLine1,
		AddressLine2:     e.AddressLine2,
		City:             e.City,
		Region:           e.Region,
		Country:          e.Country,
		Pincode:          e.Pincode,
		BoundaryCode:     e.BoundaryCode,
		Latitude:         e.Latitude,
		Longitude:        e.Longitude,
		LocationAccuracy: e.LocationAccuracy,
		RequestID:        e.RequestID,
		AuditDetail:      newAuditDetail(e.CreatedBy, e.ModifiedBy, e.CreatedTime, e.ModifiedTime),
	}
}

func AddressToEntity(d *AddressDTO) *Address {
	if d == nil {
		return nil
	}
	return &Address{
		ID:               d.ID,
		Type:             d.Type,
		DoorNo:           d.DoorNo,
		BuildingName:     d.BuildingName,
		Street:           d.Street,
		Landmark:         d.Landmark,
		AddressLine1:     d.AddressLine1,
		AddressLine2:     d.AddressLine2,
		City:             d.City,
		Region:           d.Region,
		Country:          d.Country,
		Pincode:          d.Pincode,
		BoundaryCode:     d.BoundaryCode,
		Latitude:         d.Latitude,
		Longitude:        d.Longitude,
		LocationAccuracy: d.LocationAccuracy,
		// TenantID is set by the enrichment service from the parent Individual.
		// Audit fields are set by enrichment.
	}
}
